cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -4504,39 +4500,203 @@ mod tests {
+                "must be one of `principal`, `action`, or `resource`"
+            ));
+        });
+        let p_src = "permit(resource, action, resource);";
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &e, &ExpectedErrorMessage::error("the variable `resource` is invalid in this policy scope clause, the variable `principal` is expected"));
+        });
+
+        let p_src = "permit(principal, principal, resource);";
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &e, &ExpectedErrorMessage::error("the variable `principal` is invalid in this policy scope clause, the variable `action` is expected"));
+        });
+        let p_src = "permit(principal, if, resource);";
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &e, &ExpectedErrorMessage::error_and_help(
+                "expected a variable that is valid in the policy scope; found: `if`",
+                "must be one of `principal`, `action`, or `resource`"
+            ));
+        });
+
+        let p_src = "permit(principal, action, like);";
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &e, &ExpectedErrorMessage::error_and_help(
+                "expected a variable that is valid in the policy scope; found: `like`",
+                "must be one of `principal`, `action`, or `resource`"
+            ));
+        });
+        let p_src = "permit(principal, action, principal);";
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &e, &ExpectedErrorMessage::error("the variable `principal` is invalid in this policy scope clause, the variable `resource` is expected"));
+        });
+        let p_src = "permit(principal, action, action);";
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &e, &ExpectedErrorMessage::error("the variable `action` is invalid in this policy scope clause, the variable `resource` is expected"));
+        });
+    }
+
+    #[test]
+    fn invalid_scope_operator() {
+        let p_src = r#"permit(principal > User::"alice", action, resource);"#;
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &e, &ExpectedErrorMessage::error_and_help(
+                "not a valid policy scope constraint: >",
+                "policy scope constraints must either `==`, `in`, `is`, or `_ is _ in _`"
+            ));
+        });
+        let p_src = r#"permit(principal, action != Action::"view", resource);"#;
+        assert_matches!(parse_policy_template(None, p_src), Err(e) => {
+            expect_err(p_src, &e, &ExpectedErrorMessage::error_and_help(
+                "not a valid policy scope constraint: !=",
+                "policy scope constraints must either `==`, `in`, `is`, or `_ is _ in _`"
```

Comment:
Action doesn't accept `is`. See https://github.com/cedar-policy/cedar/issues/564